internal/resources/webform/webform.js

```diff
@@ -2131,46 +2160,58 @@ window.initGRPCForm = function(services, invokeURI, metadataURI, debug) {
                     }
                     // TODO(jh): button or some other mechanism to view result as raw JSON.
                 } else {
-                    $("#grpc-response-data").hide();
+                    $("#grpc-response-responseData").hide();
                 }
 
-                if (data.error) {
+                if (responseData.error) {
+                    addHistory({
+                        ...history,
+                        duration: window.performance.now() - startTime,
+                        result: responseData.error.name ?? 'FAILED',
+                        responseData: responseData,
+                    });
                     $("#grpc-response-error").show();
-                    $("#grpc-response-error-desc").text(data.error.name);
-                    $("#grpc-response-error-num").text('(' + data.error.code + ')');
-                    if (data.error.message !== data.error.name) {
+                    $("#grpc-response-error-desc").text(responseData.error.name);
+                    $("#grpc-response-error-num").text('(' + responseData.error.code + ')');
+                    if (responseData.error.message !== responseData.error.name) {
                         var msg = $("#grpc-response-error-msg");
                         msg.show();
-                        msg.text(data.error.message);
+                        msg.text(responseData.error.message);
                     } else {
                         $("#grpc-response-error-msg").hide();
                     }
                     // TODO(jh): show error details, in the same way as we
                     // show response messages above
                     $("#grpc-response-error-details").hide();
                 } else {
+                    addHistory({
+                        ...history,
+                        duration: window.performance.now() - startTime,
+                        result: 'OK',
```

Comment:
I think `result` is strictly, and easily, derivable from other fields we are storing. How about we just omit it?